workspaces/global-floating-action-button/plugins/global-floating-action-button/src/components/FloatingButton.tsx

```diff
@@ -0,0 +1,158 @@
+    maxWidth: '150px',
+    gap: '10px',
+  },
+  bottomRight: {
+    bottom: '60px',
+    right: '60px',
+  },
+  bottomCenter: {
+    bottom: '60px',
+    left: '50%',
+  },
+}));
+
+export const FloatingButton = ({
+  floatingButtons,
+  position,
+}: {
+  floatingButtons: FloatingActionButton[];
+  position: Position;
+}) => {
+  const theme = useTheme();
+  const { pathname } = useLocation();
+  const subMenuRef = React.useRef<HTMLDivElement>(null);
+  const [isMenuOpen, setIsMenuOpen] = React.useState(false);
+  const [subMenuDirection, setSubMenuDirection] = React.useState('column');
+  const fabButton = useStyles();
+
+  React.useEffect(() => {
+    const floatingButtonElement = document.getElementById('floating-button');
+    const screenHeight = window.innerHeight;
+    if (floatingButtonElement) {
+      const { top } = floatingButtonElement?.getBoundingClientRect();
+      if (top < screenHeight / 2) {
+        setSubMenuDirection('column');
+      } else {
+        setSubMenuDirection('column-reverse');
+      }
+    }
+    return () => {
+      setIsMenuOpen(false);
+    };
+  }, [pathname]);
+
+  const fabs = React.useMemo(
+    () => filterAndSortButtons(floatingButtons, pathname),
+    [floatingButtons, pathname],
+  );
+
+  const handleClick = () => {
+    setIsMenuOpen(!isMenuOpen);
+  };
+
+  if (fabs?.length === 0) {
+    return null;
+  }
+
+  const getFloatingButton = () => {
+    if (fabs.length > 1) {
+      return (
+        <FAB
+          icon={isMenuOpen ? <CloseIcon /> : <MenuIcon />}
+          size="medium"
+          color="default"
+          label="Menu"
+          onClick={handleClick}
+          tooltip="Menu"
+        />
+      );
+    }
+
+    return (
+      <FAB
+        icon={fabs[0].icon}
+        size={fabs[0].size || 'medium'}
+        color={fabs[0].color}
+        label={fabs[0].label as string}
+        onClick={fabs[0].onClick}
+        url={fabs[0].url}
+        tooltip={fabs[0].toolTip}
+        showLabel={fabs[0].showLabel}
+      />
+    );
+  };
+  return (
+    <div
+      className={classnames(fabButton.button, fabButton[position])}
+      style={{
+        flexDirection: (subMenuDirection as any) || 'column-reverse',
+      }}
+      id="floating-button"
+      data-testId="floating-button"
+      ref={subMenuRef}
+    >
+      {getFloatingButton()}
+      {isMenuOpen && (
+        <Slide
+          container={subMenuRef.current}
+          easing={{
+            enter: theme.transitions.easing.easeOut,
+            exit: theme.transitions.easing.sharp,
+          }}
+        >
+          <>
+            {fabs?.map(fb => {
+              return (
+                <FAB
+                  icon={fb.icon}
+                  size={fb.size || 'medium'}
+                  color={fb.color}
+                  label={fb.label}
+                  onClick={fb.onClick}
+                  url={fb.url}
+                  tooltip={fb.toolTip}
+                  showLabel={fb.showLabel}
```

Comment:
Here as well. I think it will be cleaner if you pass one prop fab or action=fb.

I'm think calling everything FAB is confusing. That's why I bring up action here.